libcontainer/factory_linux.go

```diff
@@ -159,13 +159,27 @@ func (l *LinuxFactory) Create(id string, config *configs.Config) (Container, err
 	}
 	containerRoot := filepath.Join(l.Root, id)
 	if _, err := os.Stat(containerRoot); err == nil {
-		return nil, newGenericError(fmt.Errorf("Container with id exists: %v", id), IdInUse)
+		pid, err := ioutil.ReadFile(filepath.Join(containerRoot, "task"))
+		if err == nil {
+			procPath := fmt.Sprintf("/proc/%s/cmdline", pid)
+			if cmdline, err := ioutil.ReadFile(procPath); err == nil {
+				_, exeName := filepath.Split(string(cmdline))
+				exeName = exeName[0:4]
+				if exeName[0:4] == "runc" {
```

Comment:
another thought, line 167 will panic with 'slice bounds out of range' if the exeName is less than 4 characters long.